src/store.cc

```diff
@@ -418,6 +418,8 @@ StoreEntry::hashDelete()
 void
 StoreEntry::lock(const char *context)
 {
+    if (!lock_count && mem_obj && mem_obj->repl.data)
+        mem_obj->data_hdr.allowedToFreeWithReplPolicy(false);
```

Comment:
Let's simplify/streamline this as well:

```suggestion
    if (mem_obj)
        mem_obj->data_hdr.allowedToFreeWithReplPolicy(false);
```

Please also move this code right _after_ the lock is incremented, to clarify the intent/precondition/cause.